### DISCUSSION TITLE
How to import Tangrams heightmap?

### DISCUSSION THREAD

**Discussion by ljlhouben** [TokisanGames/Terrain3D#587] [Questions] [answered]
Simple question, As a newbie, I would like to import a piece of map from https://tangrams.github.io/heightmapper/ as a starting point to create a map for a game level. I tried to follow the https://terrain3d.readthedocs.io/en/stable/docs/import_export.html tutorial, but not all settings are explained in detail therefore I don't understand them all. 
1. How to use the scale-factor from the Tangrams site to setup the correct height?
2. How to setup the 'pieces/portions' and what is the 'best' size? 
3. To what size do I need to shrink the image canvas? (My original file is 7780x4395 pixels)
4. The file format from the Tangrams site is default *.png with 32 bit color depth (grayscale though), is this usable or do I need to convert?
5. If I have to convert to which file format?
So preferably a step by step guide with the latest interface would be nice :-) 

Regards

Ludo

**Comment by TokisanGames** [comment, marked as answer]
> The file format from the Tangrams site is default *.png with 32 bit color depth (grayscale though), is this usable or do I need to convert?

The document you linked explains that Godot doesn't support greater than 8-bit png. Convert to exr in photoshop, gimp, or Krita, then import that. 

If you save your file as normalized (values 0-1), then scale it on import to whatever height you want, or whatever peak height to trough you get from that website. 

> To what size do I need to shrink the image canvas? ( My original file is 7780x4395 pixels)

Max size is +/-32767 (total 65.5kx65.5k). I'm sure it will fit. You must have a large enough region size though. You get up to 32x32 regions. So 8k / 32 is 256, minimum region size (default) . 


> So preferably a step by step guide with the latest interface would be nice :-)

Sorry I can't support 3rd party tools like this tangrams app or image editing apps. Learning about image formats and how to manipulate them, understanding what format your data is in, which tools support which formats, how to adjust into new formats, all can be done with YouTube tutorials and is well worth your time. You'll learn more through testing and experimentation. These are basic skills you'll use throughout your entire career, not just in gamedev.

How to import data in our tool is already written about, and demonstrated in my first tutorial video. The differences with the latest are minor.

**Reply by ljlhouben** [reply]
Thank you for your reply. I will have a go then
